### PR TITLE
Revert "Revert "Update documentation for bootstrap token feature (#640)""

### PR DESCRIPTION
**Pull request by dharmjit**
Reverts vmware-tanzu/cluster-api-provider-bringyourownhost#653

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -107,19 +138,19 @@ $ cat /etc/hosts
 
 If you are trying this on your own hosts, then for each host
 1. Download the [byoh-hostagent-linux-amd64](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/download/v0.2.0/byoh-hostagent-linux-amd64)
```
Comment (shivi28) [inline docs/getting_started.md:140]: Shall we point this to latest release agent ?

**Review by shivi28 — COMMENTED** [review]
minor change requested
/lgtm

File: docs/getting_started.md [hunk 2]
```diff
@@ -81,6 +81,37 @@ done
...
+
+```shell
+APISERVER=$(kubectl config view -ojsonpath='{.clusters[0].cluster.server}')
+CA_CERT=$(kubectl config view --flatten -ojsonpath='{.clusters[0].cluster.certificate-authority-data}')
+```
+
+Create a BootstrapKubeconfig CR as follows
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: v1beta1
```
Comment (anusha94) [inline docs/getting_started.md:96]: I believe the apiVersion should be `infrastructure.cluster.x-k8s.io/v1beta1`

File: docs/getting_started.md (hunk 1 above)
Comment (anusha94) [reply docs/getting_started.md:140]: yes, we should

File: docs/local_dev.md [hunk 3]
```diff
@@ -78,13 +78,44 @@ Wait for all the resources to come up, status can be viewed in Tilt UI.
...
+
+```shell
+APISERVER=$(kubectl config view -ojsonpath='{.clusters[0].cluster.server}')
+CA_CERT=$(kubectl config view --flatten -ojsonpath='{.clusters[0].cluster.certificate-authority-data}')
+```
+
+Create a BootstrapKubeconfig CR as follows
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: v1beta1
```
Comment (anusha94) [inline docs/local_dev.md:92]: same - infrastructure.cluster.x-k8s.io/v1beta1

File: docs/getting_started.md [hunk 4]
```diff
@@ -81,6 +81,37 @@ done
...
+Once the BootstrapKubeconfig CR is created, fetch the object to copy the bootstrap kubeconfig file details from the Status field
+```shell
+kubectl get bootstrapkubeconfig bootstrap-kubeconfig -n default -o json | jq '.status'
+```
+Copy contents into a file called bootstrap-kubeconfig
+
+We need one bootstrap-kubeconfig per host. Create as many bootstrap-kubeconfig files as there are number of hosts (2 for this guide)
+
 ---
 ### VM Prerequisites
```
Comment (anusha94) [inline docs/getting_started.md:116]: @sachinkumarsingh092 Also, have we documented anywhere about manual approval of certs? IMO, the getting started guide should have automated flow itself. But we also need to mention about manual approval.

Comment (dharmjit) [reply docs/getting_started.md:116]: @anusha94 Currently it seems like it is not possible to provide this via `clusterctl` and the user has to download/update/apply the `infrastructure-components.yaml` for the manual approval of certs. I would suggest taking this in a patch version release. WDYT?
cc: @shivi28 @Madhur97

Comment (anusha94) [reply docs/getting_started.md:116]: Why do we need to provide via `clusterctl`? This being a feature gate / switch, can we simply ask the user to set the env variable before initializing the provider?

Comment (dharmjit) [reply docs/getting_started.md:116]: Might be missing something but how do you suggest doing this? Users need to provide the environment variable to the manager deployment hence I was thinking to pass it via `clusterctl init`.

Comment (shivi28) [reply docs/getting_started.md:116]: If it's a feature gate then we can pass it in `~/.cluster-api/clusterctl.yaml` file, in the same fashion as we do for all other feature gates variables. 
> Might be missing something but how do you suggest doing this?
For this we can through an error while creating the workload clusters that we need to set `MANUAL_CSR_APPROVAL ` feature gate. Same thing happens for clusterclass feature

Comment (shivi28) [reply docs/getting_started.md:116]: [Issue](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/668) created to handle this variable. Resolving this thread for now.

File: docs/getting_started.md (hunk 1 above)
Comment (dharmjit) [reply docs/getting_started.md]: fixed

**Review by anusha94 — APPROVED** [review]

**Pull request merged**